packages/vue-app/template/utils.js

```diff
@@ -259,7 +259,7 @@ export function promisify(fn, context) {
 
 // Imported from vue-router
 export function getLocation(base, mode) {
-  let path = window.location.pathname
+  let path = decodeURI(window.location.pathname)
```

Comment:
How about catching and processing error from decodeUri?

Reply:
normally I'd be all for it but in this case we're decoding directly from window.location.pathname. since that's been encoded by the browser we're know it's been done without error and so decoding won't throw an error either. I'd be more worried about the decoding of user-generated strings from the config such as this one:
https://github.com/nuxt/nuxt.js/blob/5331a91223c7febd2ed612e2fe7264dd9c3c5928/packages/vue-app/template/router.js#L157